perfkitbenchmarker/publisher.py

```diff
@@ -645,6 +657,92 @@ def _deDotKeys(self, res):
     return res
 
 
+class InfluxDBPublisher(SamplePublisher):
+
+  def __init__(self, influx_uri=None, influx_db_name=None):
+    # set to default above in flags unless changed
+    self.influx_uri = influx_uri
+    self.influx_db_name = influx_db_name
+
+  def PublishSamples(self, samples):
+    formated_samples = []
+    for sample in samples:
+      formated_samples.append(self._ConstructSample(sample))
+    self._Publish(formated_samples)
+
+  def _Publish(self, formated_samples):
+    try:
+      self._CreateDB()
+      body = '\n'.join(formated_samples)
+      self._WriteData(body)
+    except (IOError, httplib.HTTPException) as http_exception:
+      logging.error('Error connecting to the database:  %s', http_exception)
+
+  def _ConstructSample(self, sample):
+    timestamp = str(int((10 ** 9) * sample['timestamp']))
+    measurement = 'perfkitbenchmarker'
+
+    tag_set_metadata = ''
+    if 'metadata' in sample:
+      if sample['metadata']:
+        tag_set_metadata = ','.join(self._FormatToKeyValue(sample['metadata']))
+    tag_keys = ('test', 'official', 'owner', 'run_uri', 'sample_uri',
+                'metric', 'unit')
+    ordered_tags = collections.OrderedDict([(k, sample[k]) for k in tag_keys])
+    tag_set = ','.join(self._FormatToKeyValue(ordered_tags))
+    if tag_set_metadata:
+      tag_set += ',' + tag_set_metadata
+
+    field_set = '%s=%s' % ('value', sample['value'])
+
+    sample_constructed_body = '%s,%s %s %s' % (measurement, tag_set,
+                                               field_set, timestamp)
+    return sample_constructed_body
+
+  def _FormatToKeyValue(self, sample):
+    key_value_pairs = []
+    for k, v in sample.iteritems():
+      if v == '':
+        v = '\\"\\"'
+      v = str(v)
+      v = v.replace(',', '\,')
+      v = v.replace(' ', '\ ')
+      key_value_pairs.append('%s=%s' % (k, v))
+    return key_value_pairs
+
+  def _CreateDB(self):
```

Comment:
Hey @gareth-ferneyhough I made the requested change to remove the extra blank line, however as for why _CreateDB() is always called: this is because it is idempotent in the sense that if the DB already exists it will simply return a 200 code without re-creating it.